runtime/smithy-client/common/src/aws/smithy/kotlin/runtime/client/endpoints/AuthScheme.kt

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package aws.smithy.kotlin.runtime.client.endpoints
+
+import aws.smithy.kotlin.runtime.InternalApi
+import aws.smithy.kotlin.runtime.util.AttributeKey
+
+/**
+ * Static attribute key for AWS endpoint auth schemes.
+ */
+@InternalApi
+public val AuthSchemesAttributeKey: AttributeKey<List<AuthScheme>> = AttributeKey("authSchemes")
+
+/**
+ * A set of signing constraints for an AWS endpoint.
+ */
+@InternalApi
+public sealed class AuthScheme {
```

Comment:
**Suggestion**: I realize this is namespaced to `endpoints` package but given the identity/auth SRA changes this is kind of overloaded as a type. Is there a better name for this? I think long term endpoints will not be allowed to influence auth scheme, only signing context (i.e. IF `sigv4` is chosen as auth scheme, this additional context should be fed to the signer).

* `SigningContext` 
* `EndpointSigningContext`
* `AuthSchemeContext`